flask_app/tasks.py

```diff
@@ -121,23 +119,35 @@ def _get_os_type(ssh_client: SSHClient) -> str:
     return stdout.read().decode("utf-8").strip().lower()
 
 
-def _generate_random_combadge_name(stringLength: int) -> str:
-    random_string = ''.join(random.choice(string.ascii_lowercase) for i in range(stringLength))
+def _generate_random_combadge_name(string_length: int) -> str:
+    random_string = ''.join(random.choice(string.ascii_lowercase) for i in range(string_length))
     return f"combadge_{random_string}"
 
 
 def _upload_combadge(ssh_client: SSHClient, combadge_version: str) -> str:
     os_type = _get_os_type(ssh_client)
-    combadge_name = _generate_random_combadge_name(stringLength=10)
+    combadge_name = _generate_random_combadge_name(string_length=10)
     combadge_type_identifier = combadge_version if combadge_version == 'v1' else os_type
-    is_nt = combadge_type_identifier == 'windows'
+    is_windows = combadge_type_identifier == 'windows'
+
     local_combadge_path = get_combadge_path(combadge_type_identifier)
-    remote_combadge_path = str(PureWindowsPath(os.path.join('C:', 'tmp', f'{combadge_name}.exe'))) if is_nt else os.path.join('/tmp', combadge_name)
+    if is_windows:
+        remote_combadge_dir = str(PureWindowsPath(os.path.join('C:', 'temp')))
+        combadge_name = f'{combadge_name}.exe'
+        remote_combadge_path = str(PureWindowsPath(os.path.join(remote_combadge_dir, combadge_name)))
+    else:
+        remote_combadge_dir = '/tmp'
+        remote_combadge_path = os.path.join(remote_combadge_dir, combadge_name)
     logger.debug(f"combadge path: {remote_combadge_path}")
 
     with SFTPClient.from_transport(ssh_client.get_transport()) as sftp:
+        try:
+            sftp.chdir(remote_combadge_dir)
+        except IOError:
+            sftp.mkdir(remote_combadge_dir)
+            sftp.chdir(remote_combadge_dir)
```

Comment:
Why is it needed? You anyway don't handle the possible exception of `chdir`, and it doesn't affect subsequent operations.